components/camel-azure-storage-datalake/src/main/java/org/apache/camel/component/azure/storage/datalake/DataLakeComponent.java

```diff
@@ -0,0 +1,126 @@
+import org.apache.camel.support.DefaultComponent;
+import org.apache.camel.util.ObjectHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Component("azure-storage-datalake")
+public class DataLakeComponent extends DefaultComponent {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DataLakeComponent.class);
+
+    @Metadata(description = "configuration object for datalake")
+    private DataLakeConfiguration configuration = new DataLakeConfiguration();
+
+    public DataLakeComponent() {
+    }
+
+    public DataLakeComponent(final CamelContext camelContext) {
+        super(camelContext);
+    }
+
+    @Override
+    protected Endpoint createEndpoint(String uri, String remaining, Map<String, Object> parameters) throws Exception {
+
+        if (remaining == null || remaining.trim().isEmpty()) {
+            throw new IllegalArgumentException("At least the account name must be specified");
+        }
+
+        final DataLakeConfiguration configuration;
+        if (this.configuration != null) {
+            configuration = this.configuration.copy();
+        } else {
+            configuration = new DataLakeConfiguration();
+        }
+
+        final String[] details = remaining.split("/");
+
+        configuration.setAccountName(details[0]);
+
+        if (details.length > 1) {
+            configuration.setFileSystemName(details[1]);
+        }
+
+        final DataLakeEndpoint endpoint = new DataLakeEndpoint(uri, this, configuration);
+        setProperties(endpoint, parameters);
+        setClientFromRegistry(configuration);
+
+        setCredentialsFromRegistry(configuration);
+        validateConfiguration(configuration);
+
+        return endpoint;
+    }
+
+    public DataLakeConfiguration getConfiguration() {
+        return configuration;
+    }
+
+    public void setConfiguration(DataLakeConfiguration configuration) {
+        this.configuration = configuration;
+    }
+
+    private void setCredentialsFromRegistry(final DataLakeConfiguration configuration) {
+        final DataLakeServiceClient client = configuration.getServiceClient();
+
+        if (client == null) {
+            Set<StorageSharedKeyCredential> storageSharedKeyCredentials
+                    = getCamelContext().getRegistry().findByType(StorageSharedKeyCredential.class);
+            Set<ClientSecretCredential> clientSecretCredentials
+                    = getCamelContext().getRegistry().findByType(ClientSecretCredential.class);
+
+            if (storageSharedKeyCredentials.size() == 1) {
+                configuration.setSharedKeyCredential(storageSharedKeyCredentials.stream().findFirst().get());
+            }
+
+            if (clientSecretCredentials.size() == 1) {
+                configuration.setClientSecretCredential(clientSecretCredentials.stream().findFirst().get());
+            }
+
+        }
+    }
+
+    private void setClientFromRegistry(final DataLakeConfiguration configuration) {
```

Comment:
Actually we no longer need this, we can now autowire the client by adding `@Metadata(autowired = true)` to the serviceClient configuration and remove this. An example [here](https://github.com/apache/camel/commit/bc8fa176a66bfab4b5385305c626fddc01d29c2b#diff-fa0be9a42d8b24af5d483c6e806968d302c829087bcf363ce5fc2e67559ca863)